fhevm/params.go

```diff
@@ -214,31 +223,38 @@ func DefaultGasCosts() GasCosts {
 			tfhe.FheUint64: 43000 + AdjustFHEGas,
 		},
 		FheGetCiphertext: map[tfhe.FheUintType]uint64{
-			tfhe.FheUint8:  12000,
-			tfhe.FheUint16: 14000,
-			tfhe.FheUint32: 18000,
-			tfhe.FheUint64: 28000,
+			tfhe.FheBool:     10000,
+			tfhe.FheUint8:    12000,
+			tfhe.FheUint16:   14000,
+			tfhe.FheUint32:   18000,
+			tfhe.FheUint64:   28000,
+			tfhe.FheUint160:  50000,
+			tfhe.FheUint2048: 100000,
 		},
 		// TODO: The values here are chosen somewhat arbitrarily.
 		// Also, we don't take into account whether a ciphertext existed (either "current" or "original") for the given handle.
 		// Finally, costs are likely to change in the future.
 		FheStorageSstoreGas: map[tfhe.FheUintType]uint64{
-			tfhe.FheUint4:   SstoreFheUint4Gas,
-			tfhe.FheUint8:   SstoreFheUint4Gas * 2,
-			tfhe.FheUint16:  SstoreFheUint4Gas * 4,
-			tfhe.FheUint32:  SstoreFheUint4Gas * 8,
-			tfhe.FheUint64:  SstoreFheUint4Gas * 16,
-			tfhe.FheUint128: SstoreFheUint4Gas * 32,
-			tfhe.FheUint160: SstoreFheUint4Gas * 40,
+			tfhe.FheBool:     SstoreFheUint4Gas / 2,
+			tfhe.FheUint4:    SstoreFheUint4Gas,
+			tfhe.FheUint8:    SstoreFheUint4Gas * 2,
+			tfhe.FheUint16:   SstoreFheUint4Gas * 4,
+			tfhe.FheUint32:   SstoreFheUint4Gas * 8,
+			tfhe.FheUint64:   SstoreFheUint4Gas * 16,
+			tfhe.FheUint128:  SstoreFheUint4Gas * 32,
+			tfhe.FheUint160:  SstoreFheUint4Gas * 40,
+			tfhe.FheUint2048: SstoreFheUint4Gas * 120,
```

Comment:
shouldnt proice be proportional to number of bits here as well to avoid DDOS?

Reply:
In general, yes, but the size is around 8 MB for FheUint2048. Not sure what the best approach is here.

Reply:
I see, then if price becomes prohibitive maybe it would make sense to reduce SstoreFheUint4Gas.

Reply:
Yeah, could be. But that could still be an attack.